From f89d21fdcbef05b323021fb84e70940adafaea8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Macholda?= <tomas.macholda@nic.cz>
Date: Fri, 27 Mar 2026 15:45:34 +0100
Subject: [PATCH] base-files: uci-defaults: add ucidef_add_led_color
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add support for multi intensity LEDs.

Example:

> # cat /etc/board.d/01_leds
> ucidef_set_led_netdev "lan" "LAN" "rgb:lan" "br-lan"
> ucidef_add_led_color "lan" "red" "255"

This adds the uci 'color_red' option.

Signed-off-by: Tomáš Macholda <tomas.macholda@nic.cz>
---
 .../base-files/files/lib/functions/uci-defaults.sh | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/package/base-files/files/lib/functions/uci-defaults.sh b/package/base-files/files/lib/functions/uci-defaults.sh
index 5293ce12c0..953f48f862 100644
--- a/package/base-files/files/lib/functions/uci-defaults.sh
+++ b/package/base-files/files/lib/functions/uci-defaults.sh
@@ -428,6 +428,20 @@ _ucidef_set_led_common() {
 	json_add_string sysfs "$sysfs"
 }
 
+ucidef_add_led_color() {
+	local cfg="led_$1"
+	local channel="$2"
+	local val="$3"
+
+	json_select_object led
+	json_select_object "$1"
+
+	json_add_string "color_${channel}" "$val"
+
+	json_select ..
+	json_select ..
+}
+
 ucidef_set_led_default() {
 	local default="$4"
 
-- 
2.53.0

